src/core/ext/filters/client_channel/lb_policy/xds/xds_override_host.cc

```diff
@@ -471,77 +490,72 @@ OrphanablePtr<LoadBalancingPolicy> XdsOverrideHostLb::CreateChildPolicyLocked(
 }
 
 absl::StatusOr<ServerAddressList> XdsOverrideHostLb::UpdateAddressMap(
-    absl::StatusOr<ServerAddressList> addresses,
-    int override_host_status_mask) {
-  // It will also retain subchannels until after update. Otherwise destroying
-  // the subchannels will try to reacquire the locks in the ResetSubchannel...
-  std::map<std::string, SubchannelEntry, std::less<>> new_map;
-  absl::StatusOr<ServerAddressList> child_addresses;
+    absl::StatusOr<ServerAddressList> addresses) {
+  if (!addresses.ok()) {
+    return addresses;
+  }
+  ServerAddressList return_value;
+  std::map<const std::string, XdsHealthStatus> addresses_for_map;
+  for (const auto& address : *addresses) {
+    XdsHealthStatus status = GetAddressHealthStatus(address);
+    auto key = grpc_sockaddr_to_string(&address.address(), false);
+    if (key.ok()) {
+      addresses_for_map.insert({std::move(*key), status});
```

Comment:
I think this is incorrect.  We need to add entries even for hosts whose health status is not present in the override status set, because if the entry is not present in the map, then we won't wrap the subchannel, which means that we won't have a way to update the connectivity state later if the host's health status changes to a value that *is* in the set.

I believe the code I sent above is correct.